### PR TITLE
feat: Add overrides for excludeTargets and includeTargets

### PR DESCRIPTION
**Pull request by obs-gh-justindaines**
Adding option to override [kube-state-events](https://github.com/observeinc/kube-state-events/tree/main) container with [include-target](https://github.com/observeinc/kube-state-events/blob/6d6b1d1e1fa39f843db75e387b1d7126a68fb132/cmd/kube-state-events/main.go#L111) and [exclude-target](https://github.com/observeinc/kube-state-events/blob/6d6b1d1e1fa39f843db75e387b1d7126a68fb132/cmd/kube-state-events/main.go#L116).

Validated with following test values:
```
containerOverrides:
  excludeTargets:
    - events.k8s.io/v1/events
    - apps/v1/replicasets
  includeTargets:
    - apps/v1/deployments
```

### CODE REVIEW

File: charts/events/templates/deployment.yaml [hunk 1]
```diff
@@ -55,6 +67,7 @@ spec:
             - autoscaling/v1
```
Comment (obs-gh-abhishekrao) [inline charts/events/templates/deployment.yaml:67]: What happens when either of these three are specified in `-exclude-target` or `-include-target`?

Comment (obs-gh-justindaines) [reply charts/events/templates/deployment.yaml:67]: Adding `-include-target` doesn't add any value. I included it for possible future use. Do you think we should drop it?

I did catch an error, should be [] and not {}. I'll update accordingly, but would appreciated thoughts on omitting `includeTargets`.

Comment (obs-gh-abhishekrao) [reply charts/events/templates/deployment.yaml:67]: I think it's good to have that, but my question was different. It looks like additional APIs can be included by simply specifying `-<API>`, which you can see specified as `autoscaling/v1`, `batch/v1`, `networking.k8s.io/v1`. I was curious as to how the new flags behave with these in tandem.

**Comment by obs-gh-abhishekrao** [comment]
Also, please see if you can add these flags to the default CI tests in `charts/stack/ci/test-values.yaml`. That'll help capture if this change introduces any regressions or works as expected.

**Review by obs-gh-abhishekrao — APPROVED** [review]

**Pull request merged**